[CI] ConcurrentSeqNoVersioningIT testSeqNoCASLinearizability failing

Not sure if this failed because it was aborted or if it's a genuine linearizability failure. Hopefully the former!

**Build scan:**
https://gradle-enterprise.elastic.co/s/gx6tcdvqgpy6y/tests/:server:internalClusterTest/org.elasticsearch.versioning.ConcurrentSeqNoVersioningIT/testSeqNoCASLinearizability
**Reproduction line:**
```
./gradlew ':server:internalClusterTest' --tests "org.elasticsearch.versioning.ConcurrentSeqNoVersioningIT.testSeqNoCASLinearizability" -Dtests.seed=67F2A8CBA2C4EF6D -Dtests.locale=hu -Dtests.timezone=Etc/GMT-8 -Druntime.java=21
```

**Applicable branches:**
main

**Reproduces locally?:**
Didn't try

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.versioning.ConcurrentSeqNoVersioningIT&tests.test=testSeqNoCASLinearizability


**Failure excerpt:**
```
java.lang.AssertionError: Must be linearizable

  at __randomizedtesting.SeedInfo.seed([67F2A8CBA2C4EF6D:BC3C86A1B8D6CDD9]:0)
  at org.junit.Assert.fail(Assert.java:88)
  at org.junit.Assert.assertTrue(Assert.java:41)
  at org.elasticsearch.versioning.ConcurrentSeqNoVersioningIT$Partition.assertLinearizable(ConcurrentSeqNoVersioningIT.java:450)
  at java.lang.Iterable.forEach(Iterable.java:75)
  at org.elasticsearch.versioning.ConcurrentSeqNoVersioningIT.testSeqNoCASLinearizability(ConcurrentSeqNoVersioningIT.java:179)
  at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
  at java.lang.reflect.Method.invoke(Method.java:580)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:48)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
  at java.lang.Thread.run(Thread.java:1583)

```

--- a/server/src/internalClusterTest/java/org/elasticsearch/versioning/ConcurrentSeqNoVersioningIT.java
+++ b/server/src/internalClusterTest/java/org/elasticsearch/versioning/ConcurrentSeqNoVersioningIT.java
@@ -118,7 +118,6 @@ public class ConcurrentSeqNoVersioningIT extends AbstractDisruptionTestCase {
     // multiple threads doing CAS updates.
     // Wait up to 1 minute (+10s in thread to ensure it does not time out) for threads to complete previous round before initiating next
     // round.
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/102255")
     public void testSeqNoCASLinearizability() {
         final int disruptTimeSeconds = scaledRandomIntBetween(1, 8);
 
@@ -434,24 +433,31 @@ public class ConcurrentSeqNoVersioningIT extends AbstractDisruptionTestCase {
                 history
             );
             LinearizabilityChecker.SequentialSpec spec = new CASSequentialSpec(initialVersion);
-            boolean linearizable = false;
+            Boolean linearizable = null;
             try {
                 linearizable = LinearizabilityChecker.isLinearizable(spec, history, missingResponseGenerator());
             } catch (LinearizabilityCheckAborted e) {
-                logger.warn("linearizability check check was aborted", e);
+                logger.warn("linearizability check was aborted, assuming linearizable", e);
             } finally {
                 try {
-                    if (linearizable) {
+                    if (Boolean.TRUE.equals(linearizable)) {
                         // ensure that we can serialize all histories.
                         writeHistory(new OutputStreamStreamOutput(OutputStream.nullOutputStream()), history);
                     } else {
-                        logger.error("Linearizability check failed. Spec: {}, initial version: {}", spec, initialVersion);
+                        final var outcome = linearizable == null ? "inconclusive" : "unlinearizable";
+
+                        logger.error(
+                            "Linearizability check did not succeed. Spec: {}, initial version: {}, outcome: {}",
+                            spec,
+                            initialVersion,
+                            outcome
+                        );
                         // we dump base64 encoded data, since the nature of this test is that it does not reproduce even with same seed.
                         try (
                             var chunkedLoggingStream = ChunkedLoggingStream.create(
                                 logger,
                                 Level.ERROR,
-                                "raw unlinearizable history in partition " + id,
+                                "raw " + outcome + " history in partition " + id,
                                 ReferenceDocs.LOGGING // any old docs link will do
                             );
                             var output = new OutputStreamStreamOutput(chunkedLoggingStream)
@@ -462,20 +468,20 @@ public class ConcurrentSeqNoVersioningIT extends AbstractDisruptionTestCase {
                             var chunkedLoggingStream = ChunkedLoggingStream.create(
                                 logger,
                                 Level.ERROR,
-                                "visualisation of unlinearizable history in partition " + id,
+                                "visualisation of " + outcome + " history in partition " + id,
                                 ReferenceDocs.LOGGING // any old docs link will do
                             );
                             var writer = new OutputStreamWriter(chunkedLoggingStream, StandardCharsets.UTF_8)
                         ) {
                             LinearizabilityChecker.writeVisualisation(spec, history, missingResponseGenerator(), writer);
                         }
+                        assertNull("Must not be unlinearizable", linearizable);
                     }
                 } catch (IOException e) {
                     logger.error("failure writing out history", e);
                     fail(e);
                 }
             }
-            assertTrue("Must be linearizable", linearizable);
         }
     }
 

